src/sentry/api/endpoints/project_tagkey_values.py

```diff
@@ -40,29 +38,16 @@ def get(self, request, project, key):
         except tagstore.TagKeyNotFound:
             raise ResourceDoesNotExist
 
-        queryset = tagstore.get_tag_value_qs(
+        paginator = tagstore.get_tag_value_paginator(
             project.id,
             environment_id,
             tagkey.key,
             query=request.GET.get('query'),
+            order_by='-last_seen',
         )
 
         return self.paginate(
             request=request,
-            queryset=queryset,
-            order_by='-last_seen',
-            paginator_cls=DateTimePaginator,
-            on_results=lambda results: serialize(
-                map(  # XXX: This is a pretty big abstraction leak
-                    lambda instance: TagValue(
-                        key=instance.key,
-                        value=instance.value,
-                        times_seen=instance.times_seen,
-                        first_seen=instance.first_seen,
-                        last_seen=instance.last_seen,
-                    ),
-                    results,
-                ),
-                request.user
-            ),
+            paginator=paginator,
+            on_results=lambda results: serialize(results, request.user),
```

Comment:
I'm not sure this `on_results` call is required at this point but I don't have the energy to check